receiver.py

```diff
@@ -0,0 +1,52 @@
+from time import sleep
+from typing import Union
+
+import daiquiri
+from celery import Celery
+from celery.events.state import Task, Worker
+
+from exporters import Exporter
+from utils import is_event_type_task
+
+logger = daiquiri.getLogger(__name__)
+
+
+class CeleryEventReceiver:
+    def __init__(self, broker: str):
+        self.broker = broker
+        self.celery_app = Celery(
+            broker=self.broker,
+        )
+        self.exporters: list[Exporter] = []
+        self.state = self.celery_app.events.State()
+
+    def attach(self, exporter):
+        self.exporters.append(exporter)
+
+    def notify(self, event: Union[Task, Worker]):
+        for exporter in self.exporters:
+            exporter.process_event(event)
+
+    def notify_event(self, event):
+        event_type: str = event.get("type")
+        if not is_event_type_task(event_type):
+            return
+
+        event_details, event_type = self.state.event(event)
+        event, is_create_event = event_details
+        logger.debug(
+            f"Receiver event: {event.uuid} {event.name}, {event.state} {event.timestamp} with data: {event}"
+        )
+        self.notify(event)
+
+    def run(self) -> None:
+        with self.celery_app.connection() as connection:
+            while True:
```

Comment:
thinking out loud:
Do we require some rate limiting or sleep here? Will it hammer redis with connections trying to receive events?

Reply:
Ohh i see the sleep, but it's only on exceptions, not on success.

Reply:
add a sleep here won't actually help, because the capture (`receiver.capture`) is blocking call, which internal starts a consumer (guessing with from code with just 1 connection) and calls the handlers on receiving a new events.

i added the `while True` to handle cases when there is a exception in the handler func, even there it would reuse the above connection from line `with self.celery_app.connection() as connection` 

will keep it as such for now.